### PR TITLE
Timeout individual Ansible tasks

### LINKED ISSUE #484: Make Sonar more robust to ssh hangs
If ssh to a host hangs for any reason, Sonar should time out, record that the host timed out, and keep collecting data from other hosts and generating reports.

### PR DESCRIPTION
**Pull request by noahl**
Provide a custom ssh executable to Ansible that runs regular ssh
inside of 'timeout', and use this to add a timeout to every task. Add
infrastructure for testing this ssh executable by making it time out a
program that will hang forever. Also add a test for SSH crashes.

Closes #484 .

### CODE REVIEW

File: quipucords/scanner/network/inspect.py [hunk 1]
```diff
@@ -79,7 +84,24 @@ def run(self):
...
+                raise ScannerException(msg)
+
+            logger.info('Inspect scan task started for %s.',
+                        self.scan_task)
+            log_msg = '%d total connected, %d completed, %d'\
+                ' remaining, and make %d failed hosts'
+            logger.info(log_msg,
+                        num_total, num_completed, num_remaining, num_failed)
+
+            self.inspect_scan(connected)
```
Comment (chambridge) [inline quipucords/scanner/network/inspect.py]: What was the point in moving this block of code out of `inspect_scan` and changing the `inspect_scan` signature?  Testability?

**Comment by chambridge** [comment]
Looks like failing tests for the crash simulation.

File: quipucords/scanner/network/inspect.py (hunk 1 above)
Comment (noahl) [reply quipucords/scanner/network/inspect.py]: Yes, testability. Otherwise the test would have been much more convoluted.

**Comment by noahl** [comment]
@chambridge I'll keep working on the failing tests. The tests are failing on Travis and not my machine, and I don't know why, but it means I have a longer than usual turnaround time on testing my code.

**Pull request merged**